step-capstone/src/components/TripList.js

```diff
@@ -33,30 +38,64 @@ class TripList extends React.Component{
     .catch(error => {
       console.log("Error Getting Trips")
       console.error(error)
-    });
+    })
+  }
+
+  componentDidMount() {
+    this.loadTrips();
+    
   }
 
+
   handleOpenTrip(tripId) {
     this.setState({selectedTrip : tripId})
     sessionStorage.setItem("tripId",tripId);
   }
 
+  handleAddTrip(newTrip) {
+    console.log("Adding new trip to database...");
+    const reference = "/users/" + this.state.userId + "/trips";
+    this.context.addTrip(reference, newTrip);
+    this.loadTrips();
```

Comment:
this is something we should probably all discuss, but here and in delete you reload trips from firebase after an addition or a deletion. which works, but how we handled things in Trip was we change the local version of trips[], so that we dont have to reload and call to firebase on every change. not as big of a deal here but might be nice to decide on one approach

Reply:
good point

Reply:
Updated (After Zach and I discussed): To avoid reload the whole things, we need to manually give the trip its ID. We can do that, but it requires changing a lot of thing. Probably we can do so in the different PR.